Fix deprecatedFallback getting called when auto-filtering is enabled (#2557)
GetStream/stream-chat-swift: CHANGELOG.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).

# Upcoming
## StreamChat

### 🐞 Fixed

> - Fix channel list auto-filtering issue in cases where we are querying for channels that the user may not be a member [#2557](https://github.com/GetStream/stream-chat-swift/pull/2557)

Reviewer: Maybe we should say what is the cause of the issue (otherwise, "issue" is a bit too generic)
```suggestion
- Fix channel disappearing when channel list auto-filtering is enabled and the user is not a member of the channel [#2557](https://github.com/GetStream/stream-chat-swift/pull/2557)
```
